### PR TITLE
Base page

### PR DESCRIPTION
**Pull request by Tbotny**

### CODE REVIEW

File: src/test/java/page/AbstractBasePage.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package page;
+
+import org.openqa.selenium.WebDriver;
+
+public abstract class AbstractBasePage {
+    private final WebDriver driver;
```
Comment (light-di) [inline src/test/java/page/AbstractBasePage.java]: delete

File: src/test/java/driver/DriverInstance.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class DriverInstance {
+    public DriverInstance() {
+    }
+
+    private static WebDriver driver;
```
Comment (light-di) [inline src/test/java/driver/DriverInstance.java]: public

File: src/test/java/driver/DriverInstance.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package driver;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class DriverInstance {
```
Comment (light-di) [inline src/test/java/driver/DriverInstance.java]: DriverSingleTone

File: src/test/java/page/AbstractBasePage.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    public TopNavBar topNavBar;
+    public Header header;
+    public CategoryMenu categoryMenu;
+    public Footer footer;
```
Comment (light-di) [inline src/test/java/page/AbstractBasePage.java]: Change to AbstractContent

File: src/test/java/page/CategoryMenu.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.pagefactory.ElementLocatorFactory;
+
+public class CategoryMenu extends AbstractBasePage {
+
+
+    protected CategoryMenu(WebDriver driver) {
```
Comment (light-di) [inline src/test/java/page/CategoryMenu.java]: delete

File: src/test/java/page/CategoryMenu.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.pagefactory.ElementLocatorFactory;
+
+public class CategoryMenu extends AbstractBasePage {
```
Comment (light-di) [inline src/test/java/page/CategoryMenu.java]: delete extends

File: src/test/java/page/CategoryMenu.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+
+    protected CategoryMenu(WebDriver driver) {
+        super(driver);
+
+    }
+
+    AbstractBasePage openPage() {
```
Comment (light-di) [inline src/test/java/page/CategoryMenu.java]: delete

File: src/test/java/page/Footer.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package page;
```
Comment (light-di) [inline src/test/java/page/Footer.java]: change to AbstractContent class

File: src/test/java/page/AbstractBasePage.java (hunk 4 above)
Comment (light-di) [inline src/test/java/page/AbstractBasePage.java]: Add private String pageAddress;

File: src/test/java/page/AbstractBasePage.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+
+
+    abstract AbstractBasePage openPage();
```
Comment (light-di) [inline src/test/java/page/AbstractBasePage.java]: public void openPage() {
DriverInstance.driver.get(this.pageAddress);
}

**Review by light-di — APPROVED** [review]

**Review by light-di — APPROVED** [review]

**Review by TolkachDima — APPROVED** [review]

**Review by light-di — APPROVED** [review]

**Review by TolkachDima — APPROVED** [review]

**Pull request merged**